newIDE/app/src/InstancesEditor/TileSetVisualizer.js

```diff
@@ -0,0 +1,640 @@
+        const mouseY = event.clientY - bounds.top + 1;
+        const { x, y } = getGridCoordinatesFromPointerCoordinates({
+          pointerX: mouseX,
+          pointerY: mouseY,
+          columnCount,
+          rowCount,
+          displayedTileSize,
+        });
+        if (!allowMultipleSelection) {
+          if (
+            tileMapTileSelection &&
+            tileMapTileSelection.kind === 'single' &&
+            tileMapTileSelection.coordinates.x === x &&
+            tileMapTileSelection.coordinates.y === y
+          ) {
+            onSelectTileMapTile(null);
+          } else {
+            onSelectTileMapTile({
+              kind: 'single',
+              coordinates: { x, y },
+              flipHorizontally: shouldFlipHorizontally,
+              flipVertically: shouldFlipVertically,
+            });
+          }
+          return;
+        }
+        if (!clickStartCoordinates) return;
+
+        const {
+          x: startX,
+          y: startY,
+        } = getGridCoordinatesFromPointerCoordinates({
+          pointerX: clickStartCoordinates.x,
+          pointerY: clickStartCoordinates.y,
+          columnCount,
+          rowCount,
+          displayedTileSize,
+        });
+        const newSelection =
+          tileMapTileSelection && tileMapTileSelection.kind === 'multiple'
+            ? { ...tileMapTileSelection }
+            : { kind: 'multiple', coordinates: [] };
+        if (startX === x && startY === y) {
+          if (
+            tileMapTileSelection &&
+            // Click on a tile.
+            tileMapTileSelection.kind === 'multiple'
+          ) {
+            addOrRemoveCoordinatesInArray(newSelection.coordinates, {
+              x,
+              y,
+            });
+          }
+        } else {
+          for (
+            let columnIndex = Math.min(startX, x);
+            columnIndex <= Math.max(startX, x);
+            columnIndex++
+          ) {
+            for (
+              let rowIndex = Math.min(startY, y);
+              rowIndex <= Math.max(startY, y);
+              rowIndex++
+            ) {
+              if (newSelection && newSelection.kind === 'multiple') {
+                addOrRemoveCoordinatesInArray(newSelection.coordinates, {
+                  x: columnIndex,
+                  y: rowIndex,
+                });
+              }
+            }
+          }
+        }
+        onSelectTileMapTile(newSelection);
+      } finally {
+        setClickStartCoordinates(null);
+        setRectangularSelectionTilePreview(null);
+      }
+    },
+    [
+      displayedTileSize,
+      columnCount,
+      rowCount,
+      tileMapTileSelection,
+      onSelectTileMapTile,
+      shouldFlipHorizontally,
+      shouldFlipVertically,
+      allowMultipleSelection,
+      clickStartCoordinates,
+    ]
+  );
+
+  React.useEffect(
+    () => {
+      // On dismount, remove tile map selection.
+      return () => onSelectTileMapTile(null);
```

Comment:
shouldn't this be the responsibility of the top component to remove which one was selected when this component is not displayed anymore?
It feels weird having it this way around

Reply:
Hmm, indeed, I'll see to move this up.